src/sentry/tasks/commit_context.py

```diff
@@ -177,17 +177,42 @@ def process_commit_context(
             author_to_user = get_users_for_authors(commit.organization_id, authors)
             user_dct: Mapping[str, Any] = author_to_user.get(str(commit.author_id), {})
 
-            group_owner, created = GroupOwner.objects.update_or_create(
```

Comment:
i think you can still use `update_or_create` here? something like:

```
group_owner, created = GroupOwner.objects.update_or_create(
    group_id=group_id,
    type=GroupOwnerType.SUSPECT_COMMIT.value,
    user_id=user_dct.get("id"),
    project=project,
    organization_id=project.organization_id,
    context__contains={"commitId": commit.id},
    defaults={
        "date_added": django_timezone.now(),
        "context": {  # This would REPLACE existing context!
            "commitId": commit.id,
            "suspectCommitStrategy": SuspectCommitStrategy.SCM_BASED.value,
        }
    }
)
```